drltrace/drltrace.c

```diff
@@ -104,6 +105,134 @@ static app_pc exe_start;
 /* runtest.cmake assumes this is the prefix, so update both when changing it */
 #define STDERR_PREFIX "~~~~ "
 
+/* Table that maps syscall names to syscall_info_t structure */
+#define NAME2ENTRY_TABLE_HASH_BITS 13 /* 1.5K of them, x2 for no-prefix entries + Zw */
+static hashtable_t name2entry;
+
+/* Table that maps address (found when dll is loading) to the syscall_info_t structure */
+#define ADDR2ENTRY_TABLE_HASH_BITS 8
+static hashtable_t addr2entry_table;
+
+/* XXX: We need to add support of printing from secondary table (e.g. ioctl) (xref i#1549) */
+#ifdef WINDOWS
+extern dr_os_version_info_t win_ver; /* already defined in drsyscall_windows.c */
```

Comment:
None of these are exported by the drsyscall library, though.  They're not part of the library interface!  All uses here should be through interfaces.

It sounds like you could add a static drsys_lookup_syscall() routine that takes in a name and returns a drsys_syscall_t "handle", right?  Then there's no need to break the library's abstractions in multiple ways here by using its internal, private data.